.github/workflows/main.yml

```diff
@@ -1,5 +1,11 @@
 name: Main
-on: push
+on:
+  push:
+    branches:
+      - main
+  pull_request:
```

Comment:
I believe this should be `pull_request_target`, otherwise it won't run on PRs from forked repositories. Not sure `branches` is used either; you could copy-paste the whole trigger from `pr.yml`.

Reply:
Docs say that that [`pull_request_target`](https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#pull_request_target) runs the action from the target (i.e. `origin/master` in most circumstances) instead of from the head of the PR's branch, so it wouldn't be testing the code in the PR.